### ISSUE TITLE
issue.transition options maybe wrong?

### ISSUE THREAD

**Issue by sindhus** [floralvikings/jira-connector#107]
Hi,

thanks for making this software! 

I was reading through issue.transition and think that [the documentation stating that transition in opts must be a string](https://github.com/floralvikings/jira-connector/blob/master/api/issue.js#L711) could be wrong?

The documentation on Atlassian says transition is an object expecting an id key within it. 

![screen shot 2018-07-12 at 12 15 18 pm](https://user-images.githubusercontent.com/4291156/42616845-485b96ec-85cd-11e8-9a46-828ec3346b6d.png)

from [here](https://developer.atlassian.com/cloud/jira/platform/rest/) (sorry couldn't find the bookmark position in the page)

please feel free to close the issue if am wrong.

**Comment by TheTedAdams** [comment]
Fields is also supposed to be an object, but the generic `buildRequestOptions()` method expects any fields opt to be an array of strings to add to the querystring
```js
if (opts.fields) {
  qs.fields = '';
  opts.fields.forEach(function (field) {
    qs.fields += field + ','
  });
}
```

**Comment by floralvikings** [comment]
This repository is only _very_ lightly maintained; you are welcome to submit a pull request if there is behavior you'd like to add, I do occasionally check the PR list and merge code that is within the scope of this tool and doesn't include breaking changes.  

If you have found a bug please indicate the desired behavior and provide a minimal example demonstrating the incorrect behavior; I do try to fix bugs as they come up but I have very little time to devote to this repository anymore, so I can't make any guarantees.

If you have a feature you'd like to request, please indicate the desired behavior.  I will almost certainly never have the time to implement it myself but this repository does see occasional PR activity so it may be added by a generous contributor.  You are, of course, welcome to submit a pull request yourself implementing the desired behavior.

**Comment by MrRefactoring** [comment]
Solved in 2.12.0 version
